Create method to map JIT module to (source, constant) and back.

qihqi opened the pull request:
Summary:
implemented function to generate source as ExtraFilesMap and constants

  wrote function to construct jit module given (ivalue, source,
  constant) tripple.

Test Plan: unittest

Differential Revision: D34803945

zhxchen17 commented on test/cpp/jit/test_save_load.cpp:
```diff
@@ -13,6 +15,20 @@
 
+namespace {
+
+static Module roundtripThroughMobile(const Module& m) {
```
nit: we don't need both namespace {} and static

pavithranrao reviewed (approved):
LGTM

qihqi committed:
Create method to map JIT module to (source, constant) and back. (#74119)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/74119

  implemented function to generate source as ExtraFilesMap and constants

  wrote function to construct jit module given (ivalue, source,
  constant) tripple.

Test Plan: unittest

Reviewed By: pavithranrao

Differential Revision: D34803945

fbshipit-source-id: 90683da2a72708295189545a62520cd135608c2e
